crypto/ml_kem/ml_kem_ipd_ref_common/cbd.c

```diff
@@ -105,17 +103,15 @@ static void cbd3(poly *r, const uint8_t buf[3*KYBER_N/4])
     }
   }
 }
-#endif
 
-void poly_cbd_eta1(poly *r, const uint8_t buf[KYBER_ETA1*KYBER_N/4])
+void poly_cbd_eta1(ml_kem_params *params, poly *r, const uint8_t *buf)
 {
-#if KYBER_ETA1 == 2
-  cbd2(r, buf);
-#elif KYBER_ETA1 == 3
-  cbd3(r, buf);
-#else
-#error "This implementation requires eta1 in {2,3}"
-#endif
+  assert((params->eta1 == 2) || (params->eta1 == 3));
+  if (params->eta1 == 2) {
+    cbd2(r, buf);
+  } else {
+    cbd3(r, buf);
```

Comment:
This `else` is risky because release builds may remove the above `assert` so this code path may be reached by code with `eta1 != 3`. Might be worth replacing with `else if (params->eta1 == 3)` and do a separate `else` that is a no-op.

Reply:
^ ore move the assert to that `else` clause.

Reply:
Wouldn't the `assert` still be a no-op in release builds?